### PR TITLE
add queries and update

### PR DESCRIPTION
**Pull request by Mucyosoda**
vet_clinic DATABASE

>Database vet_clinic has been created with one table, and a table with 6 columns (id, name, date_of_birth, escape_attemps, neutered, weight_kg, species) and inserted rows and Update rows.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -3,4 +3,11 @@
 INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attemps) VALUES ('Agumon', date '2020-02-03', 10.23, true, 0);
 INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attemps) VALUES ('Gabumon', date '2018-11-15', 8, true, 2);
 INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attemps) VALUES ('Pikachu', date '2021-01-07', 15.04, false, 1);
-INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attemps) VALUES ('Devimon', date '2017-05-12', 11, true, 5);
\ No newline at end of file
+INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attemps) VALUES ('Devimon', date '2017-05-12', 11, true, 5);
```
Comment (ZeenatLawal) [inline data.sql]: - [ ] Kindly remove the `DATE` text written before the date_of_birth entries, the datatype has already been declared in the creation of the table.

File: queries.sql [hunk 2]
```diff
@@ -9,3 +9,36 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <>'Gabumon';
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
 
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
Comment (ZeenatLawal) [inline queries.sql:14]: - [ ] Verify that the change was made before and after rollback as stated in the requirements. e.g
```sql
BEGIN;
UPDATE animals SET species = 'unspecified';
SELECT species from animals; -- verify that change was made
ROLLBACK;
SELECT species from animals; -- verify that change was undone
```

File: queries.sql [hunk 3]
```diff
@@ -9,3 +9,36 @@ SELECT * FROM animals WHERE neutered = true;
...
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species ='';
+COMMIT;
```
Comment (ZeenatLawal) [inline queries.sql:19]: - [ ] Verify that the change was made before and after the commit is made as stated in the requirements. e.g
```sql
BEGIN;
UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

SELECT species from animals; -- verify that change was made
COMMIT;
SELECT species from animals; -- verify that change persists after commit
```

File: queries.sql [hunk 4]
```diff
@@ -9,3 +9,36 @@ SELECT * FROM animals WHERE neutered = true;
...
+
+BEGIN
+vet_clinic=*# delete from animals;
+DELETE 11
+ROLLBACK;
```
Comment (ZeenatLawal) [inline queries.sql]: - [ ] Verify that the change was made before and after the rollback is made as stated in the requirements. e.g
```sql
BEGIN;
DELETE * FROM animals;
SELECT COUNT(*) FROM ANIMALS;

ROLLBACK;
SELECT COUNT(*) FROM ANIMALS;
```

File: queries.sql [hunk 5]
```diff
@@ -9,3 +9,36 @@ SELECT * FROM animals WHERE neutered = true;
...
+SAVEPOINT animalsavepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT animalsavepoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attemps = 0;
+SELECT ROUND(AVG(weight_kg),2) FROM animals;
+SELECT name, escape_attemps FROM animals ORDER BY escape_attemps DESC LIMIT 1;
```
Comment (ZeenatLawal) [inline queries.sql]: - [ ] To find who escapes the most, neutered or not neutered animals? Kindly get the avg of the number of escape attempts and group it by neutered. Check example below:
```sql
 SELECT neutered, AVG(escape_attempts) FROM animals GROUP BY neutered;
```

File: queries.sql [hunk 6]
```diff
@@ -9,3 +9,36 @@ SELECT * FROM animals WHERE neutered = true;
...
+SELECT MAX(weight_kg) FROM animals WHERE species = 'digimon';
+SELECT Min(weight_kg) FROM animals WHERE species = 'digimon';
+SELECT Min(weight_kg) FROM animals WHERE species = 'pokemon';
+SELECT Max(weight_kg) FROM animals WHERE species = 'pokemon';
```
Comment (ZeenatLawal) [inline queries.sql]: - [ ] To find the minimum and maximum weight of each type of animal, it'll be good to get the data with one statement as shown below.
```sql
SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
```

File: queries.sql [hunk 7]
```diff
@@ -9,3 +9,36 @@ SELECT * FROM animals WHERE neutered = true;
...
+SELECT AVG(escape_attemps) FROM animals WHERE (date_of_birth BETWEEN '1990-01-01' AND '2000-01-01') AND (species = 'pokemon');
+SELECT AVG(escape_attemps) FROM animals WHERE (date_of_birth BETWEEN '1990-01-01' AND '2000-01-01') AND (species = 'digimon');
```
Comment (ZeenatLawal) [inline queries.sql]: - [ ] To find the average number of escape attempts per animal type of those born between 1990 and 2000, it'll be good to get the data with one statement as shown below.
```sql
SELECT species, AVG(escape_attempts) FROM animals GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/31/12';
```

**Review by ZeenatLawal — CHANGES_REQUESTED** [review]

**Review by descholar-ceo — APPROVED** [review]
Hi @Mucyosoda , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**